### PR TITLE
Added an algorithm to calculate the present value of cash flows

### PR DESCRIPTION
**Pull request by sahilg13**
### Describe your change:
I added a financial algorithm that calculates the present value of a stream of yearly cash flows at a given discount rate.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: financial/present_value.py
Comment (rohan472000) [inline financial/present_value.py:22]: There are a few improvements that can be made to the `present_value` function to make it more robust and avoid potential errors:

1. Validate that the cash flows list is not empty before calculating the present value.

       if not cash_flows:
            raise ValueError("Cash flows list cannot be empty")

2. Validate that the discount rate is not negative, as negative discount rates don't make sense in finance.

       if discount_rate < 0:
        raise ValueError("Discount rate cannot be negative")

File: financial/present_value.py
Comment (rohan472000) [inline financial/present_value.py:32]: Round the present value to a certain number of decimal places for consistency.

    decimal_places = 2
    return round(present_value, decimal_places )

**Review by rohan472000 — CHANGES_REQUESTED** [review]

File: financial/present_value.py
Comment (sahilg13) [reply financial/present_value.py:26]: Thank you for these improvements, I have updated the code.

File: financial/present_value.py
Comment (sahilg13) [reply financial/present_value.py:40]: Thank you for these improvements, I have updated the code.

**Review by rohan472000 — APPROVED** [review]

**Comment by rohan472000** [comment]
To get your PR merge kindly ask a review from @cclauss as he has write access.

**Comment by sahilg13** [comment]
@cclauss Hi! If you could please review and merge this PR that would be great!

File: financial/present_value.py
Comment (cclauss) [inline financial/present_value.py:27]: Please decare variables close to where they are first used.
```suggestion
```

File: financial/present_value.py
Comment (cclauss) [inline financial/present_value.py:36]:
```suggestion
    present_value = sum(cash_flow / ((1 + discount_rate) ** i)  for i, cash_flow in enumerate(cash_flows))
```

File: financial/present_value.py
Comment (cclauss) [inline financial/present_value.py:39]:
```suggestion
    return round(present_value, ndigits=2)
```

**Comment by sahilg13** [comment]
@cclauss Thank you for your improvements. I have committed the suggested changes and I am ready for another review.

File: financial/present_value.py
Comment (cclauss) [inline financial/present_value.py:7]: Docstrings are better because they will appear in automated documentation like Sphinx, etc.
```suggestion
"""
Reference: https://www.investopedia.com/terms/p/presentvalue.asp

An algorithm that calculates the present value of a stream of yearly cash flows given...
1. The discount rate (as a decimal, not a percent)
2. An array of cash flows, with the index of the cash flow being the associated year

Note: This algorithm assumes that cash flows are paid at the end of the specified year
```

File: financial/present_value.py
Comment (cclauss) [inline financial/present_value.py:38]:
```suggestion
    if discount_rate < 0:
        raise ValueError("Discount rate cannot be negative")
    if not cash_flows:
        raise ValueError("Cash flows list cannot be empty")
    present_value = sum(
        cash_flow / ((1 + discount_rate) ** i) for i, cash_flow in enumerate(cash_flows)
    )
    return round(present_value, ndigits=2)
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**
